local/src/main/scala/com/salesforce/op/local/OpWorkflowModelLocal.scala

```diff
@@ -85,29 +74,26 @@ trait OpWorkflowModelLocal extends Serializable {
     /**
      * Prepares a score function for local scoring
      *
-     * @param spark Spark Session needed for preparing scoring function,
-     *              Once scoring function is returned the Spark Session can be shutdown
-     *              since it's not required during local scoring.
      * @return score function for local scoring
      */
-    def scoreFunction(implicit spark: SparkSession): ScoreFunction = {
+    def scoreFunction: ScoreFunction = {
       // Prepare the stages for scoring
       val stagesWithIndex = model.getStages().zipWithIndex
 
-      // Prepare an empty DataFrame with transformed schema & metadata (needed for loading MLeap models)
-      val transformedData = makeTransformedDataFrame(model)
-
       // Collect all OP stages
+      // todo fix wrapped predictors
       val opStages = stagesWithIndex.collect { case (s: OpTransformer, i) => OPModel(s.getOutputFeatureName, s) -> i }
 
       // Collect all Spark wrapped stages
-      val sparkStages = stagesWithIndex.filterNot(_._1.isInstanceOf[OpTransformer]).collect {
-        case (opStage: OPStage with SparkWrapperParams[_], i) if opStage.getSparkMlStage().isDefined =>
-          val sparkStage = opStage.getSparkMlStage().get.asInstanceOf[Transformer]
-          (opStage, sparkStage, i)
+      val mleapStages = stagesWithIndex.filterNot(_._1.isInstanceOf[OpTransformer]).collect {
+        case (opStage: OPStage with SparkWrapperParams[_], i) if opStage.getLocalMlStage().isDefined =>
+          val model = opStage.getLocalMlStage().get
```

Comment:
Better pattern match and error gracefully when local model is missing. For example:
```scala
opStage.getLocalMlStage() match {
  case None => throw new RuntimeException("Local model not found for stage ${opStage.uid} of type ${opStage.getClass}")
  case Some(model) =>
     // Apply model 
}


```